internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -244,10 +244,15 @@ func (r *NumaflowControllerRolloutReconciler) reconcile(
 		r.recorder.Eventf(controllerRollout, corev1.EventTypeNormal, "Deleting", "Deleting NumaflowControllerRollout")
 		if controllerutil.ContainsFinalizer(controllerRollout, finalizerName) {
 			ppnd.GetPauseModule().DeletePauseRequest(controllerKey)
+			// Check if dependent resources are deleted, if not then requeue after 5 seconds
+			if !r.isDependentResourceDeleted(ctx, controllerRollout) {
+				numaLogger.Warn("Dependent resources are not deleted yet, requeue after 5 seconds")
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
nit: wonder if instead of setting `RequeueAfter`, we could just set `Requeue: true`? Then we should be able to make use of the exponential backoff algorithm